hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyDataStreamOutput.java

```diff
@@ -81,6 +81,14 @@ enum StreamAction {
 
   private long clientID;
 
+  /**
+   * Indicates if an atomic write is required. When set to true,
+   * the amount of data written must match the declared size during the commit.
+   * A mismatch will prevent the commit from succeeding.
+   * This is essential for operations like S3 put to ensure atomicity.
+   */
+  private boolean requiresAtomicWrite;
```

Comment:
I would suggest change the requiresAtomicWrite to atomicKeyCreation.

Reply:
Done